src/main/java/io/vlingo/http/resource/Configuration.java

```diff
@@ -0,0 +1,121 @@
+  }
+
+  public Configuration with(final Timing timing) {
+    this.timing = timing;
+    return this;
+  }
+
+  public int port() {
+    return this.port;
+  }
+
+  public Sizing sizing() {
+    return this.sizing;
+  }
+
+  public Timing timing() {
+    return this.timing;
+  }
+
+  private Configuration() {
+    this.port = 8080;
+    this.sizing = Sizing.define();
+    this.timing = new Timing(10, 10, 100);
+  }
+
+  private Configuration(final Properties properties) {
+    this();
+
+    this.port = Integer.parseInt(properties.getProperty("server.http.port", String.valueOf(this.port)));
+    final int dispatcherPoolSize = Integer.parseInt(properties.getProperty("server.dispatcher.pool", String.valueOf(this.sizing.dispatcherPoolSize)));
+    final int maxBufferPoolSize = Integer.parseInt(properties.getProperty("server.buffer.pool.size", String.valueOf(this.sizing.maxBufferPoolSize)));
+    final int maxMessageSize = Integer.parseInt(properties.getProperty("server.message.buffer.size", String.valueOf(this.sizing.maxMessageSize)));
+    final int probeInterval = Integer.parseInt(properties.getProperty("server.probe.interval", String.valueOf(this.timing.probeInterval)));
+    final long probeTimeout = Long.parseLong(properties.getProperty("server.probe.timeout", String.valueOf(this.timing.probeTimeout)));
+    final long requestMissingContentTimeout = Long.parseLong(properties.getProperty("server.request.missing.content.timeout", String.valueOf(this.timing.requestMissingContentTimeout)));
+
+    this.sizing = new Sizing(dispatcherPoolSize, maxBufferPoolSize, maxMessageSize);
+    this.timing = new Timing(probeInterval, probeTimeout, requestMissingContentTimeout);
+  }
+
+  public static class Sizing {
+    public final int dispatcherPoolSize;
+    public final int maxBufferPoolSize;
+    public final int maxMessageSize;
+
+    public Sizing(final int dispatcherPoolSize, final int maxBufferPoolSize, final int maxMessageSize) {
+      this.dispatcherPoolSize = dispatcherPoolSize;
+      this.maxBufferPoolSize = maxBufferPoolSize;
+      this.maxMessageSize = maxMessageSize;
+    }
+
+    public static Sizing define() {
+      return new Sizing(10, 100, 65535);
+    }
+
+    public Sizing withDispatcherPoolSize(final int dispatcherPoolSize) {
+      return new Sizing(dispatcherPoolSize, this.maxBufferPoolSize, this.maxMessageSize);
+    }
+
+    public Sizing withMaxBufferPoolSize(final int maxBufferPoolSize) {
+      return new Sizing(this.dispatcherPoolSize, maxBufferPoolSize, this.maxMessageSize);
+    }
+
+    public Sizing withMaxMessageSize(final int maxMessageSize) {
+      return new Sizing(this.dispatcherPoolSize, this.maxBufferPoolSize, maxMessageSize);
+    }
+  }
+
+  public static class Timing {
+    public final int probeInterval;
+    public final long probeTimeout;
+    public final long requestMissingContentTimeout;
+
+    public Timing(final int probeInterval, final long probeTimeout, final long requestMissingContentTimeout) {
```

Comment:
I maintained the constructor public because of [ServerTest](https://github.com/vlingo/vlingo-http/blob/master/src/test/java/io/vlingo/http/resource/ServerTest.java#L109). I see no problem with the constructor public.